lib/cn/cn.c

```diff
@@ -933,58 +935,65 @@ cn_open(
     }
 
     ctx.cn = cn;
-    ctx.nodemap = map_create(cn->cp->fanout);
+    ctx.nodemap = map_create(CN_FANOUT_MAX);
     if (ev(!ctx.nodemap)) {
         err = merr(ENOMEM);
         goto err_exit;
     }
 
     map_insert_ptr(ctx.nodemap, 0, cn->cn_tree->ct_root);
 
-    /* [HSE_REVISIT] Delete this block once we have splits. cn_kvset_cb() will populate the nodemap,
-     * and then for each node in the map, fetch the max key and call route_map_insert() with it.
-     */
-    {
-        struct ekey_generator *egen;
+    err = cndb_cn_instantiate(cndb, cnid, &ctx, (void *)cn_kvset_cb);
+    if (ev(err))
+        goto err_exit;
 
-        egen = ekgen_create(kvs_name, cn->cp);
-        assert(egen);
+    if (cn->cn_tree->ct_route_map) {
+        char kbuf[HSE_KVS_KEY_LEN_MAX];
+        struct cn_tree_node *tn, *next;
+        struct route_node *rn;
+        uint klen;
 
-        for (uint i = 0; i < cn->cp->fanout; i++) {
-            struct cn_tree_node *tn;
+        list_for_each_entry_reverse_safe(tn, next, &cn->cn_tree->ct_nodes, tn_link) {
```

Comment:
Why reverse iterate?

Reply:
Cuz I couldn't find the forward iteration macro  Lemme check again...

Reply:
Fixed.

Reply:
Nice